make/git.mk

```diff
@@ -12,3 +12,19 @@ gitver:
 	@echo "Is tagged release: \"$(IS_TAGGED_RELEASE)\""
 	@echo "Is prerelease:     \"$(IS_PRERELEASE)\""
 	@echo "Git commit hash:   \"$(GITCOMMIT)\""
+
+# Lists all remote tags on the upstream, which gives tags in format:
+# "<commit> ref/tags/<tag>". Strips commit + tag prefix, filters out tags for v1+,
+# and manually removes v1.2.0-alpha.1, since that version's manifest contains
+# duplicate CRD resources (2 CRDs with the same name) which in turn can cause problems
+# with the versionchecker test.
+# Open question: how do we decide when to refresh this target?
+bin/scratch/git/upstream-tags.txt: | bin/scratch/git
+	git ls-remote --tags --refs https://github.com/cert-manager/cert-manager.git | \
+		awk '{print $$2;}' | \
+		sed 's/refs\/tags\///' | \
+		sed -n '/v1.0.0/,$$p' | \
+		grep -v "v1.2.0-alpha.1" > $@
```

Comment:
note: see the bazel [original](https://github.com/cert-manager/cert-manager/blob/10c5d72279ae3b23f670338045c51ab10e808aa5/test/integration/versionchecker/testdata/BUILD.bazel#L2-L7) for comparison.

Bazel calls out to a fetch script while we do the downloads in make. The fetch script ignores `v1.2.0-alpha.1` [in code](https://github.com/cert-manager/cert-manager/blob/10c5d72279ae3b23f670338045c51ab10e808aa5/test/integration/versionchecker/testdata/fetch.sh#L54-L56) whereas we ignore it here.

Reply:
The trouble with running every time is it's a remote network request and so can fail / hang / induce flakiness in local development. I just ran `git ls-remote --tags --refs https://github.com/cert-manager/cert-manager.git` of times and it took 880ms, 2000ms, 1300ms, 1660ms. That's an unacceptably large overhead to me.

I think I'm kinda OK with it as-is because our presubmits will run this every time (since they start from a clean checkout). So even if you have an out-of-date file locally, you'd notice when you raise a PR and the tests fail.

Reply:
> So even if you have an out-of-date file locally, you'd notice when you raise a PR and the tests fail

Would it then make sense to log somewhere something along the lines of 'looking at tags 'x,y,z' or something along those lines to make it easier for a person who's trying to understand why this failed in CI, but not locally?

Reply:
I think this is a great suggestion; I've added a [commit](https://github.com/cert-manager/cert-manager/pull/4865/commits/dadb2aa8a2cb104f420c6305f08faf10d1ed224f) to this PR to achieve this; does it look alright to you?